### DISCUSSION TITLE
Tokens detected by PyMuPDF are not visible on the PDF page

### DISCUSSION THREAD

**Discussion by rudra0713** [pymupdf/PyMuPDF#3030] [Q&A]
Hi, I have been experimenting with PyMuPDF recently and I started by drawing boxes around the detected tokens. For one of the pages in my PDF document, I noticed that some of the returned token boxes are completely empty in the pdf but PyMuPDF was showing some tokens corresponding to those boxes.

I have drawn a subset of the returned token boundaries on the pdf page and attached the image here. Between tokens "2020" and "2019", there are a bunch of green boxes that according to PyMyPDF contain tokens (for example, "x", "y", "z"). Later, when I opened the PDF in a PDF editor and searched for those tokens ("x", "y", "z"), the PDF editor found those tokens inside those boxes. 

From this experiment, I think PyMuPDF is doing a good job of determining the tokens, but for my downstream task, I would like to discard tokens that are not visible on the PDF page. Is there a way to achieve this?
![small_font_12](https://github.com/pymupdf/PyMuPDF/assets/26213283/c6413872-8fb8-4775-96ef-85e8b908ca05)

I am also attaching the original pdf, the image shown here is page 20 of the pdf.
[Teck-2022-Sustainability-Report.pdf](https://github.com/pymupdf/PyMuPDF/files/13993257/Teck-2022-Sustainability-Report.pdf)

**Comment by JorjMcKie** [comment]
This is no issue but a typical Discussions item. Converting.

**Comment by JorjMcKie** [comment]
You did not provide the file in question. An image is no help for clarifying what is going on.
There are more than one possible reason for why this happens: the text may white on white background, or behind the pages background color, be written with no rendering, etc.
Impossible to tell like that.

**Comment by rudra0713** [comment]
Hi @JorjMcKie, sorry for the late response. I had to resolve some proprietary issues regarding the pdf first. I have uploaded the pdf in my original post.

**Reply by JorjMcKie** [reply]
Sorry: still not downloadable, just points to the URL of this issue 😒.

**Reply by rudra0713** [reply]
Sorry again, can you kindly recheck now?

**Comment by JorjMcKie** [comment]
Well, these boxes
![image](https://github.com/pymupdf/PyMuPDF/assets/8290722/3f39c535-81a8-4bf8-9c88-ace5d3cc1784)
are **_not_** empty: copy and paste delivers:
![image](https://github.com/pymupdf/PyMuPDF/assets/8290722/b0a4ba6a-76e7-4409-8194-f93ed0a9e783)

Obviously, it was the PDF creator's intention to it have it exactly like that.

**Reply by rudra0713** [reply]
Yes, I agree with you, but since the tokens themselves are not visible on the surface of the PDF, is there a way to achieve the same result when extracting the token boundaries?

**Comment by JorjMcKie** [comment]
After investigating several alternatives for hiding objects, I must admit: I see no way to ignore these items on this page.
Sorry about that.

**Reply by rudra0713** [reply]
Hi @JorjMcKie, thanks a lot for letting me know.
